### ISSUE TITLE
ADAPT_DOMAIN does not take into account the specified port

### ISSUE THREAD

**Issue by peauc** [gristlabs/grist-core#1319]
### Describe the current behavior

When using adapt domain there is an issue when two used domains are using different schemes. 

Im self hosting grist on a private infra.
I'm communicating in http between servers and also using https when accessing grist over the internet. 

I cannot get grist to work using adapt domain. It tries to send back packets to the scheme configured in GRIST_HOME_URL. 
When setting this env variable to empty. it then use the env variable PORT which is also wrong.

### Steps to reproduce

Host a grist using docker. Have it being reachable over two domains using both https & http. 

You won't be able to use the two routers

### Describe the expected behavior

It should adapt to the schema passed in the host header

### Where have you encountered this bug?

- [ ] On [docs.getgrist.com](https://docs.getgrist.com)
- [ ] On a self-hosted instance

### Instance information (when self-hosting only)

* Grist instance:
  * Version:
  * URL (if it's OK for you to share it):
  * Installation mode: docker/kubernetes/...
  * Architecture: single-worker/multi-workers

* Browser name, version and platforms on which you could reproduce the bug:
* Link to browser console log if relevant:
* Link to server log if relevant:

**Comment by paulfitz** [comment]
@fflorent this report reminds me of a fix you make in https://github.com/gristlabs/grist-core/pull/1247

**Comment by fflorent** [comment]
Indeed. I guess you specifically talk about this line: https://github.com/gristlabs/grist-core/pull/1247/commits/efdf400a001389e13c5691321106cae66b3ea69f#diff-fc1da3a0a5d0e38041484575c01e5c8af0a44c54b5bdcf120ccbf6771e333d61R43

@peauc Cloning the repository, fetching my branch and building a docker image is something that you can do? If so, I would like to know if this PR fixes your issue: https://github.com/gristlabs/grist-core/pull/1247

Otherwise I can create an unofficial image for you so you can test.

**Comment by peauc** [comment]
Hello @fflorent, I would appreciate if you could build the image. It would be way easier that way

**Comment by fflorent** [comment]
I failed to build the docker image, I think it is related to the recent changes in Dockerfile:

```
 ---> e59a558778ca
Step 36/55 : COPY --from=collector-py2 /usr/local/lib/libffi.so.6* /usr/local/lib
When using COPY with more than one source file, the destination must be a directory and end with a /
```

**Comment by peauc** [comment]
Alrighty @fflorent, i'll be watching this ticket then :D. We've found a dirty workaround for the meantime
